sqlglot/optimizer/simplify.py

```diff
@@ -369,6 +370,51 @@ def absorb_and_eliminate(expression, root=True):
     return expression
 
 
+def propagate_constants(expression, root=True):
+    """
+    Propagate constants for conjunctions in DNF:
+
+    SELECT * FROM t WHERE a = b AND b = 5 becomes
+    SELECT * FROM t WHERE a = 5 AND b = 5
+
+    Reference: https://www.sqlite.org/optoverview.html
+    """
+    from sqlglot.optimizer.normalize import normalized
+
+    if (
+        isinstance(expression, exp.And)
+        and (root or not expression.same_parent)
+        and normalized(expression, dnf=True)
+    ):
+        constant_mapping: t.Dict[exp.Column, t.Tuple[int, exp.Literal]] = {}
+        for eq in expression.find_all(exp.EQ):
+            l, r = eq.left, eq.right
+
+            if isinstance(l, exp.Column) and isinstance(r, exp.Literal):
```

Comment:
We should have some kind of helper for literals, since we often treat things like `CAST('2012-01-01' AS DATE)` as a literal. Maybe a problem for another day.

Reply:
Makes sense, was thinking about this too